language-support/ts/daml-types/index.ts

```diff
@@ -79,42 +120,54 @@ export const Unit: Serializable<Unit> = {
 export type Bool = boolean;
 
 /**
- * Companion object of the `Bool` type.
+ * Companion object of the [[Bool]] type.
  */
 export const Bool: Serializable<Bool> = {
   decoder: jtv.boolean,
 }
 
 /**
- * The counterpart of DAML's `Int` type. We represent `Int`s as string in order
- * to avoid a loss of precision.
+ * The counterpart of DAML's `Int` type.
+ *
+ * We represent `Int`s as string in order to avoid a loss of precision.
  */
 export type Int = string;
 
 /**
- * Companion object of the `Int` type.
+ * Companion object of the [[Int]] type.
  */
 export const Int: Serializable<Int> = {
   decoder: jtv.string,
 }
 
 /**
- * The counterpart of DAML's `Numeric` type. We represent `Numeric`s as string
- * in order to avoid a loss of precision. The string must match the regular
- * expression `-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?`.
+ * The counterpart of DAML's `Numeric` type.
+ *
+ * We represent `Numeric`s as string in order to avoid a loss of precision. The string must match
+ * the regular expression `-?(?:0|[1-9]\d*)(?:\.\d+)?(?:[eE][+-]?\d+)?`.
```

Comment:
Do we actually validate this on the client side anywhere or do you only get the error from the JSON API. If the latter, would it make sense to introduce validation somewhere to get more readable errors?

Reply:
I don't know, @hurryabit might know ?

Reply:
Currently, we only validate data coming from the JSON API. We only validate that each `Numeric` is a string and nothing beyond that.

Long term, we should also validate the data we send to the JSON API to be able to produce better error messages if we were to send malformed data.